### ISSUE TITLE
Failed to verify signature

### ISSUE THREAD

**Issue by nikhithabyreddy** [terraform-linters/tflint#2209]
### Summary

We have a in house plugin created to check for tags in resources so that is being installed as part of tflint init  while performing the tflint --init we are getting the below error 
14:28:05 signature.go:135: [DEBUG] Failed to verify signature: failed to verify log inclusion: not enough verified log entries from transparency log: 0 < 1
Failed to install a plugin; Failed to check checksums.txt signature: failed to verify log inclusion: not enough verified log entries from transparency log: 0 < 1

### Command

tflint --init

### Terraform Configuration

```terraform
terraform {
  required_version = "1.9.3"
```

### TFLint Configuration

```HCL
plugin "xxxx" {
  enabled = true
  version = "0.1.0"
  source  = "github.com/xxxx/tflint-ruleset-xxxx"
}
```

### Output

```Shell
14:28:05 signature.go:135: [DEBUG] Failed to verify signature: failed to verify log inclusion: not enough verified log entries from transparency log: 0 < 1
Failed to install a plugin; Failed to check checksums.txt signature: failed to verify log inclusion: not enough verified log entries from transparency log: 0 < 1
```

### TFLint Version

0.55.0

### Terraform Version

1.9.3

### Operating System

- [X] Linux
- [ ] macOS
- [ ] Windows

**Comment by wata727** [comment]
Thank you for reporting this. This is due to [plugin keyless verification](https://github.com/terraform-linters/tflint/pull/2159) introduced in v0.55.
From the error, it looks like the transparency logs aren't included in the artifact attestation, but I'm not sure what cases that would happen.

Can you share the artifact attestation of the plugin you have installed?
You can see them in https://github.com/xxxx/tflint-ruleset-xxxx/attestations

**Comment by nikhithabyreddy** [comment]
@wata727  Here is the artifact attestation 
[attestation-sigstore (2).json](https://github.com/user-attachments/files/18427806/attestation-sigstore.2.json)

**Comment by nikhithabyreddy** [comment]
@wata727  Any update on the above issue ?

**Comment by wata727** [comment]
After some research, it appears that attestations built in private repositories do not include transparency logs.
https://docs.github.com/en/actions/security-for-github-actions/using-artifact-attestations/using-artifact-attestations-to-establish-provenance-for-builds#about-using-sigstore-for-artifact-attestations

Just like with GitHub CLI, we'll need to switch verification policies.
https://github.com/cli/cli/blob/v2.65.0/pkg/cmd/attestation/verification/sigstore.go#L282
https://github.com/terraform-linters/tflint/blob/v0.55.0/plugin/signature.go#L98-L103

**Comment by wata727** [comment]
It seems that we need GitHub Enterprise Cloud plan to test Artifact Attestations in private repositories.
https://docs.github.com/en/actions/security-for-github-actions/using-artifact-attestations/using-artifact-attestations-to-establish-provenance-for-builds

Supporting artifact attestations in private repositories can be difficult as we don't have sufficient funding to test the feature.
Plugins built in private repositories are likely secure enough, so skipping verification here might be a good start.

**Comment by bendrucker** [comment]
It seems like it should be possible to verify them as long as the transparency log option is off in the verifier. This does make sense, it's a private repository, it can't be identified in transparency logs.

I'm a GitHub Enterprise member so I can do one-time testing. We wouldn't have automated integration testing but it seems like the API difference is explicit enough that it won't be too hard to support. I would tend to agree that in practice external open source orgs are probably higher risk than internal compromises. But the sort of organization that wants to keep their plugins private is also more generally risk-averse and would want to defend against both.

**Comment by bendrucker** [comment]
Yeah for sure, definitely patch it for the short term until we can test verification on a private repo.

**Comment by ricardo-kh** [comment]
Hi, sorry for kicking an already closed issue.
But I think the issue is still there.

In #2223 a patch was introduced that bypasses the verification for private repositories.

https://github.com/terraform-linters/tflint/blob/f845d2a3462fbd034120695d52d8dd0b4bcf5733/plugin/install.go#L169-L173

And then in #2224 it _looks_ like that bypass was made undone.
Because it now returns `false` directly, not processing any further `if !skipVerify {` logic.

https://github.com/terraform-linters/tflint/blob/e3e1fe44431ac513b2a21ef2b6e808550502b42b/plugin/install.go#L206-L210

Which causes `verified` to always be false in that case.
Resulting in failure here: https://github.com/terraform-linters/tflint/blob/e599327729c2b79fdcc4a6a438466cca9034390d/plugin/install.go#L170-L172

#2223 was not released separately from #2224. Thus I cannot select a specific patch version to bypass this. 

If you like I can try to suggest a simple code change to "re-activate" the bypass again?
